src/visualization.ts

```diff
@@ -95,24 +97,40 @@ export class GraphVisualizer {
                             .attr('height', d.relSize * this.config.circleRadiusUnit * 2)
                     )
 
-                    // @ts-ignore
-                    const tip = d3Tip()
-                        .attr('class', 'd3-tip')
-                        .offset([-10, 0])
-                        .html((event: any, d: Node) => {
-                            return d.description;
-                        })
-                    d3.select(this.container).call(tip);
+                    let tip = new Tooltip();
+                    document.body.appendChild(tip.getNode());
 
                     let that = this;
+                    let keep = false;
+                    let linger = false;
+                    let lingerTimeout:NodeJS.Timeout;
+                    let showTimeout:NodeJS.Timeout;
+
+                    tip.getNode().onmouseover = (() => {
+                        linger = true;
+                        clearTimeout(lingerTimeout)
+                    });
+                    tip.getNode().onmouseout = ( () => {
+                        linger = false;
+                        lingerTimeout = setTimeout(this.hideTask, 800)
+                    });
+
                     nodeEnter.filter((d: Node) => {
                         return d.description !== undefined && d.description.length > 0;
                     }).on('mouseover', function (...args) {
+                        keep = true;
+                        clearTimeout(showTimeout)
+                        tip.setData(...args, that.config.circleRadiusUnit);
```

Comment:
why do we need this?

Reply:
resolved in [remove redundant code](https://github.com/Shayan-P/Shayan-P.github.io/pull/1/commits/dc7f7d314af40ddba7af6722497771ade6564a1f)